content/blog/highly-available-nfs-workload-on-hpe-greenlake-for-private-cloud-enterprise-using-serviceguard-for-linux.md

```diff
@@ -0,0 +1,748 @@
+
+## Assign Roles to API Client
+
+Once your API Client has been created, you need to assign a Role and a Space. You can assign a Role and a Space by clicking on your new API Client and then clicking the Create Assignment button. 
+Since intent is to use this API Client to create resources in the Virtual Machines Service, we need to assign an appropriate Virtual Machines Role. Choose a Role like ‘Private Cloud Tenant Contributor’ and choose the same Space as used earlier, I.e., ‘Default.’ 
+
+Note: More details on HPE GreenLake user roles can be found in the HPE GreenLake documentation.
+
+Set API Client Usernames and Passwords
+
+When a user creates virtual machines using the HPE GreenLake for Private Cloud Enterprise: Virtual Machines user interface, they first set the Linux and Windows username and password. Once this is done, any virtual machines subsequently created by that user will inherit these credentials. The user can later use these credentials to log into these virtual machines.
+API Clients which are used to create virtual machines can also set Linux and Windows username and password values. Since the API Client does not use the HPE GreenLake for Private Cloud Enterprise: Virtual Machines user interface, this must be done via an API call.
+Here is a sample script which reads the VM\*USERNAME and VM_PASSWORD environment variables and uses the values for Linux and Windows username and password for the API Client. The script assumes a Location value of ‘FTC06’ and Space value of ‘Default’. 
+To execute this script, first set appropriate values for the VM_USERNAME and VM_PASSWORD environment variables. Next, execute the resource file, which was created earlier, which sets the HPEGL\** environment variables for your API Client. Finally, execute the script below.
+
+```shell
+#!/bin/bash
+export LOCATION='FTC06'
+export SPACE='Default'
+export SPACE_ENCODED=$(echo -n -e "$SPACE" | od -An -tx1 | tr ' ' % | xargs printf "%s")
+
+ACCESS_TOKEN=$(curl -s -k -X POST \
+  "${HPEGL_IAM_SERVICE_URL}/v1/token" \
+  -H "Content-Type: application/x-www-form-urlencoded" \
+  -d "client_id=${HPEGL_USER_ID}" \
+  -d "client_secret=${HPEGL_USER_SECRET}" \
+  -d grant_type=client_credentials \
+  -d scope=hpe-tenant | jq -r '.access_token')
+echo "Token: ${ACCESS_TOKEN}"
+
+curl -s -k -X GET \
+   "https://client.greenlake.hpe.com/api/iac-vmaas/v1/whoami?space=${SPACE_ENCODED}&location=${LOCATION}" \
+   -H "Authorization: ${ACCESS_TOKEN}" | jq '.'
+   
+# Sets user settings
+curl -s -k -X POST \
+  "https://client.greenlake.hpe.com/api/iac-vmaas/v1beta1/user-settings?space=${SPACE_ENCODED}&location=${LOCATION}" \
+  -H "Authorization: ${ACCESS_TOKEN}" \
+  -H "Content-Type: application/json" \
+  -d '{
+    "user": {
+      "linuxUsername": '${VM_USERNAME}',
+      "linuxPassword": '${VM_PASSWORD}',
+      "windowsUsername": '${VM_USERNAME}',
+      "windowsPassword": '${VM_PASSWORD}'
+    }
+  }'
+```
+
+Querying for infrastructure components
```

Comment:
You may want to highlight in bold the sentence and add a colon "**Querying for infrastructure components:**"